server/storage/errors.go

```diff
@@ -40,3 +40,13 @@ type ErrNoKey struct {
 func (err ErrNoKey) Error() string {
 	return fmt.Sprintf("Error, no timestamp key found for %s", err.gun)
 }
+
+// ErrBadChangeID indicates the change ID provided by the user is not
+// valid for some reason, i.e. it is out of bounds
+type ErrBadChangeID struct {
+	id string
+}
+
+func (err ErrBadChangeID) Error() string {
```

Comment:
Given the change ID is an index, I treated it as essentially "there's nothing in the range you requested", and returned an empty result rather than an error. Especially in the database cases, it's actually much easier to do that than run extra queries to determine if the ID is within the existing range.